packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergWebViewActivity.java

```diff
@@ -204,25 +214,61 @@ public void onPageFinished(WebView view, String url) {
                 injectWPBarsCssScript = removeWhiteSpace(removeNewLines(injectWPBarsCssScript));
                 evaluateJavaScript(String.format(INJECT_CSS_SCRIPT_TEMPLATE, injectWPBarsCssScript));
 
-                final Handler handler = new Handler();
-                handler.postDelayed(() -> {
-                    String preventAutosaves = getFileContentFromAssets("gutenberg-web-single-block/prevent-autosaves.js");
-                    evaluateJavaScript(preventAutosaves);
+                String injectGutenbergObserver = getFileContentFromAssets("gutenberg-web-single-block/gutenberg-observer.js");
+                evaluateJavaScript(injectGutenbergObserver);
+            }
+        });
+    }
 
-                    String insertBlock = getFileContentFromAssets("gutenberg-web-single-block/insert-block.js").replace("%@","%s");
-                    String blockContent = getIntent().getExtras().getString(ARG_BLOCK_CONTENT);
-                    insertBlock = String.format(insertBlock, blockContent);
-                    evaluateJavaScript(removeNewLines(insertBlock.replace("\\n", "\\\\n")));
+    private void evaluateJavaScript(String script) {
+        mWebView.evaluateJavascript(script, value ->
+                AppLog.e(AppLog.T.EDITOR, value));
+    }
 
-                    view.setVisibility(View.VISIBLE);
-                }, 2000);
-            }
+    private void onGutenbergReady() {
+        preventAutoSavesScript();
+        insertBlockScript();
+        final Handler handler = new Handler();
+        handler.postDelayed(() -> {
+            // We want to make sure that page is loaded
+            // with all elements before executing external JS
+            injectOnGutenbergReadyExternalSources();
+            mForegroundView.postDelayed(() -> mForegroundView.setVisibility(View.INVISIBLE), 500);
```

Comment:
I'm guessing this 500ms delay is to give the UI a chance to settle before presenting it to the user? Is there a comment that could help provide the "why" for this change? Just a thought because I liked the comment you added above on line 233, and I wondered if something similar could be helpful here.

Reply:
Now it's added. It should be clear what is the purpose.